README.md

```diff
@@ -13,13 +18,171 @@ idseq_dag examples/host_filter_dag.json
+      "gsnap.m8",
+      "gsnap.deduped.m8",
+      "gsnap.hitsummary.tab",
+      "gsnap_counts.json"
+    ],
+    "rapsearch2_out": [
+      "rapsearch2.m8",
+      "rapsearch2.deduped.m8",
+      "rapsearch2.hitsummary.tab",
+      "rapsearch2_counts.json"
+    ],
+    "taxon_count_out": ["taxon_counts.json"],
+    "annotated_out": ["annotated_merged.fa", "unidentified.fa"]
+  },
+  "steps": [
+    {
+      "in": ["host_filter_out"],
+      "out": "gsnap_out",
+      "class": "PipelineStepRunAlignmentRemotely",
+      "module": "idseq_dag.steps.run_alignment_remotely",
+      "additional_files": {
+        "lineage_db": "s3://idseq-database/taxonomy/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/taxid-lineages.db",
+        "accession2taxid_db": "s3://idseq-database/alignment_data/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/accession2taxid.db"
+
+          ,"deuterostome_db": "s3://idseq-database/taxonomy/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/deuterostome_taxids.txt"
+
+      },
+      "additional_attributes": {
+        "service": "gsnap",
+        "chunks_in_flight": 32,
+        "chunk_size": 15000,
+        "max_concurrent": 3,
+        "environment": "prod"
+      }
+    },
+    {
+      "in": ["host_filter_out"],
+      "out": "rapsearch2_out",
+      "class": "PipelineStepRunAlignmentRemotely",
+      "module": "idseq_dag.steps.run_alignment_remotely",
+      "additional_files": {
+        "lineage_db": "s3://idseq-database/taxonomy/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/taxid-lineages.db",
+        "accession2taxid_db": "s3://idseq-database/alignment_data/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/accession2taxid.db"
+
+          ,"deuterostome_db": "s3://idseq-database/taxonomy/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/deuterostome_taxids.txt"
+
+      },
+      "additional_attributes": {
+        "service": "rapsearch2",
+        "chunks_in_flight": 32,
+        "chunk_size": 10000,
+        "max_concurrent": 6,
+        "environment": "prod"
+      }
+    },
+    {
+      "in": ["gsnap_out", "rapsearch2_out"],
+      "out": "taxon_count_out",
+      "class": "PipelineStepCombineTaxonCounts",
+      "module": "idseq_dag.steps.combine_taxon_counts",
+      "additional_files": {},
+      "additional_attributes": {}
+    },
+    {
+      "in": ["host_filter_out", "gsnap_out", "rapsearch2_out"],
+      "out": "annotated_out",
+      "class": "PipelineStepGenerateAnnotatedFasta",
+      "module": "idseq_dag.steps.generate_annotated_fasta",
+      "additional_files": {},
+      "additional_attributes": {}
+    }
+  ],
+  "given_targets": {
+    "host_filter_out": {
+      "s3_dir": "s3://idseq-samples-prod/samples/12/5815/results_test/2.4"
+    }
+  }
+}
+
+```
+
+### Design
+
+idseq-dag embraced simplicity. There are only two major components for dag execution:
```

Comment:
"embraced simplicity" -> "follows the KISS design principle" :)

Reply:
sure.